control-plane/tests/tests/nexus.rs

```diff
@@ -4,22 +4,200 @@ pub mod common;
+}
+
+#[actix_rt::test]
+async fn create_nexus_local_replica() {
+    let size = 10 * 1024 * 1024;
     let cluster = ClusterBuilder::builder()
         .with_pools(1)
-        .with_replicas(2, 5 * 1024 * 1024, v0::Protocol::Off)
+        .with_replicas(1, size, v0::Protocol::Off)
         .build()
         .await
         .unwrap();
 
+    let replica = format!("bdev:///{}", Cluster::replica(0, 0));
     cluster
         .rest_v0()
         .create_nexus(v0::CreateNexus {
             node: cluster.node(0),
             uuid: v0::NexusId::new(),
-            size: 10 * 1024 * 1024,
-            children: vec!["malloc:///disk?size_mb=100".into()],
+            size,
+            children: vec![replica.into()],
+        })
+        .await
+        .unwrap();
+}
+
+#[actix_rt::test]
+async fn create_nexus_replicas() {
+    let size = 10 * 1024 * 1024;
+    let cluster = ClusterBuilder::builder()
+        .with_pools(1)
+        .with_replicas(1, size, v0::Protocol::Off)
+        .with_mayastors(2)
+        .build()
+        .await
+        .unwrap();
+
+    let local = format!("bdev:///{}", Cluster::replica(0, 0));
+    let remote = cluster
+        .rest_v0()
+        .share_replica(v0::ShareReplica {
+            node: cluster.node(1),
+            pool: cluster.pool(1, 0),
+            uuid: Cluster::replica(0, 0),
+            protocol: v0::Protocol::Nvmf,
+        })
+        .await
+        .unwrap();
+
+    cluster
+        .rest_v0()
+        .create_nexus(v0::CreateNexus {
+            node: cluster.node(0),
+            uuid: v0::NexusId::new(),
+            size,
+            children: vec![local.into(), remote.into()],
+        })
+        .await
+        .unwrap();
+}
+
+#[actix_rt::test]
+async fn create_nexus_replica_not_available() {
+    let size = 10 * 1024 * 1024;
+    let cluster = ClusterBuilder::builder()
+        .with_pools(1)
+        .with_replicas(1, size, v0::Protocol::Off)
+        .with_mayastors(2)
+        .build()
+        .await
+        .unwrap();
+
+    let local = format!("bdev:///{}", Cluster::replica(0, 0));
+    let remote = cluster
+        .rest_v0()
+        .share_replica(v0::ShareReplica {
+            node: cluster.node(1),
+            pool: cluster.pool(1, 0),
+            uuid: Cluster::replica(0, 0),
+            protocol: v0::Protocol::Nvmf,
+        })
+        .await
+        .unwrap();
+    cluster
+        .rest_v0()
+        .share_replica(v0::ShareReplica {
+            node: cluster.node(1),
+            pool: cluster.pool(1, 0),
+            uuid: Cluster::replica(0, 0),
+            protocol: v0::Protocol::Off,
         })
         .await
         .unwrap();
+
+    cluster
+        .rest_v0()
+        .create_nexus(v0::CreateNexus {
+            node: cluster.node(0),
+            uuid: v0::NexusId::new(),
+            size,
+            children: vec![local.into(), remote.into()],
+        })
+        .await
+        .expect_err("One replica is not present so nexus can't be created");
```

Comment:
I think `can't` should be replaced with `shouldn't` as this error message will only be shown if the nexus is erroneously created.